plugins/in_calyptia_fleet/in_calyptia_fleet.c

```diff
@@ -1830,9 +1844,17 @@ int get_calyptia_fleet_config(struct flb_in_calyptia_fleet_config *ctx)
     flb_sds_destroy(hdrname);
 
     /* create the base file. */
-    ret = get_calyptia_file(ctx, ctx->fleet_url, header, NULL, &time_last_modified);
+    ret = get_calyptia_file(ctx, ctx->fleet_url, header, NULL, &time_last_modified, &http_status);
     flb_sds_destroy(header);
 
+    /* Handle the fleet not existing */
+    if (http_status == 404) {
+        flb_plg_error(ctx->ins, "fleet no longer exists, fleet_id: %s",
+                     ctx->fleet_id ? ctx->fleet_id : "unknown");
+        ctx->fleet_deleted = FLB_TRUE;
```

Comment:
`ctx->fleet_deleted` is initialized to false and is set to true when the fleet does not exist. It never transitions back to `FLB_FALSE` after this point.